php/Snoopy.class.inc

```diff
@@ -654,9 +658,17 @@ class Snoopy
 			$port = $this->port;
 		}
 		$this->status = 0;
-		if(function_exists("stream_socket_client") && !is_null($this->IP))
```

Comment:
We can't create the stream unless `$this->IP` is not `null`.

Reply:
So it should be?

```
if(function_exists("stream_socket_client") && !is_null($this->IP))
{
  // same as previous, no change
  $opts = array('socket' => array('bindto' => $this->IP.':0'));
  $context = stream_context_create($opts);
  $fp = stream_socket_client("tcp://".$host.":".$port, $errno, $errstr, $this->_fp_timeout, STREAM_CLIENT_CONNECT, $context);
}
elseif(function_exists("stream_socket_client") && $this->useIpv4)
{
  // new section
  $opts = array('socket' => array('bindto' => $this->'0:0'));
  $context = stream_context_create($opts);
  $fp = stream_socket_client("tcp://".$host.":".$port, $errno, $errstr, $this->_fp_timeout, STREAM_CLIENT_CONNECT, $context);
}
else
  // same as previous, no change
  $fp = fsockopen($host,$port,$errno,$errstr,$this->_fp_timeout);
```

I had thought `stream_socket_client` was functionally identical to `fsockopen`, it just allowed for specifying additional options with stream_context_create. So it made sense to just group everything up, and prefer the newer method.